core/dbt/include/global_project/macros/materializations/models/materialized_view/materialized_view.sql

```diff
@@ -0,0 +1,71 @@
+{%- materialization materialized_view, default -%}
+  {% set full_refresh_mode = should_full_refresh() %}
+  {%- set existing_relation = load_cached_relation(this) -%}
+  {%- set target_relation = this.incorporate(type='view') -%}
+  {%- set intermediate_relation =  make_intermediate_relation(target_relation) -%}
+
+  -- the intermediate_relation should not already exist in the database; get_relation
+  -- will return None in that case. Otherwise, we get a relation that we can drop
+  -- later, before we try to use this name for the current operation
+  {%- set preexisting_intermediate_relation = load_cached_relation(intermediate_relation) -%}
+  /*
+     This relation (probably) doesn't exist yet. If it does exist, it's a leftover from
+     a previous run, and we're going to try to drop it immediately. At the end of this
+     materialization, we're going to rename the "existing_relation" to this identifier,
+     and then we're going to drop it. In order to make sure we run the correct one of:
+       - drop view ...
+       - drop table ...
+     We need to set the type of this relation to be the type of the existing_relation, if it exists,
+     or else "view" as a sane default if it does not. Note that if the existing_relation does not
+     exist, then there is nothing to move out of the way and subsequentally drop. In that case,
+     this relation will be effectively unused.
+  */
+  {%- set backup_relation_type = 'view' if existing_relation is none else existing_relation.type -%}
+  {%- set backup_relation = make_backup_relation(target_relation, backup_relation_type) -%}
+  -- as above, the backup_relation should not already exist
+  {%- set preexisting_backup_relation = load_cached_relation(backup_relation) -%}
+  -- grab current tables grants config for comparision later on
+  {% set grant_config = config.get('grants') %}
+
+  {{ run_hooks(pre_hooks, inside_transaction=False) }}
+
+  -- drop the temp relations if they exist already in the database
+  {{ drop_relation_if_exists(preexisting_intermediate_relation) }}
+  {{ drop_relation_if_exists(preexisting_backup_relation) }}
+
+  -- `BEGIN` happens here:
+  {{ run_hooks(pre_hooks, inside_transaction=True) }}
+
+  -- cleanup
+  -- move the existing view out of the way
+  {% if existing_relation is none %}
```

Comment:
Referencing @Fleid's direction here for convenience:

> For the full refresh, I think we should try to, in order:
> - CREATE OR REPLACE - if CREATE OR REPLACE is supported
> - CREATE, DROP, ALTER NAME - if rename is supported
> - DROP, CREATE - if none of the above
> 
> For on_configuration_changes, we should try to, in order:
> - ALTER in place
> - Apply the full refresh order

This is going to vary by adapter; but I'd like to avoid creating versions of this materialization in adapters if possible. Hence I think we should adopt something like the strategies approach that we did with the incremental materialization. In other words, we have "replace" strategy, an "alter" strategy, a "refresh" strategy, etc. In some cases, those strategies may point to the same macros (e.g. create_materialized_view_as()); but that would allow maintainers to update just the piece that is specific to that platform without needing to maintain the rest of the materialization.